plugins/registry/handler.go

```diff
@@ -282,6 +312,21 @@ func writeStatusInternalError(req *http.Request, w http.ResponseWriter) {
 	}
 }
 
+func writeStatusNotFoundError(req *http.Request, w http.ResponseWriter, error string) {
+	if req.ProtoMajor == 2 && strings.Contains(req.Header.Get("Content-Type"), "application/grpc") {
+		w.Header().Set("Content-Type", "application/grpc")
+		w.Header().Set("Trailer", "Grpc-Status")
+		w.Header().Add("Trailer", "Grpc-Message")
+		w.Header().Set("Grpc-Status", "5")
+		w.Header().Set("Grpc-Message", "NOT_FOUND")
+	} else {
+		w.WriteHeader(http.StatusNotFound)
+		w.Header().Set("Content-Type", "application/json")
+	}
+
+	fmt.Fprintln(w, error)
+}
```

Comment:
💡 Most of this is copied across statuses, at some point we could refactor the code to take into account only the grpc and HTTP status

Reply:
good point! will do that now